### ISSUE TITLE
to do: Polling to get changes from app or other service like Alexa

### ISSUE THREAD

**Issue by apexad** [apexad/homebridge-mysmartblinds-bridge#8]
**Is your feature request related to a problem? Please describe.**
The problem is that if blinds are changed using the MySmartBlinds app or other service like Alexa, they are not updated in HomeKit.

**Describe the solution you'd like**
Some kind of polling needs to be implemented so that HomeKit picks up these changes

**Describe alternatives you've considered**
Polling seems to be the only answer

**Additional context**
Moved from README to-do section
Investigate API limits and implement in a way that will be safe

**Comment by nigetech** [comment]
This is definitely key, mine just fails to respond all the time.

**Comment by apexad** [comment]
> This is definitely key, mine just fails to respond all the time.

Are you changing the blinds outside of the Home app? If so, why? I figured this issue to be low priority since once most people have their blinds setup in the Home app that would move any automations there.

If that is not the case for you, explain why to me?

**Comment by nigetech** [comment]
Yes, I use smart switches in each room. I do take your point around moving the automations to Home though, will amend this for one room and test. I also seem to get no battery percentages, just shows 0% for each blind, do you know why this would be?

**Comment by apexad** [comment]
See if moving automations to Home app helps that 0% issue. My guess is the issues could be related. MySmartBlinds API actually gives a negative battery percentage sometimes when multiple connections ask for battery. Might have to do something where I poll battery again until it gives an actual 0 or a positive number.

**Comment by apexad** [comment]
@nigetech any update?  Did moving automations to home help?  See battery percent now?

**Comment by apexad** [comment]
@nigetech one other thing about the battery... are you getting a low battery warning on the blinds reporting 0%? If the answer is no that definitely means that it is reporting a negative number.  I may have to look into this more.  Maybe it reports a negative number when it is charging or dis-charging? I'll add an issue for this, but still need your response.

**Comment by nigetech** [comment]
@apexad Just had a proper chance to play with this. Yes thank you moving all automations to Homekit helped, although I now have a further issue in that after setting up all my automations etc installing an update wiped them all out (see raised issue) Low battery issue seemed to be that it wasn't close enough to the bridge and I have repositioned my bridge as a result. I can now see and use the battery percentage, although doesn't seem to report when charging. A polling option would still be a huge plus for me even if every 10 mins or something, as if I use the blind cord or Alexa to open the blinds they're then showing as incorrect in Homekit. 

Thanks again.

**Comment by apexad** [comment]
They won’t report when charging. Not sure if the MySmartBlinds API even supports that, but if I do get around to adding polling I can probably compare the previous percent to current and update the charging characteristic.

**Comment by nigetech** [comment]
Also, another random Q - I couldn't find a way to message you separately outside of the platform. Have you ever heard of a device by Athom called Homey? It is a home automation piece of hardware which works on plugins written in Node.JS as far as I can see. Repositories are on GitHub but I don't know enough about this. I have two pieces of hardware, one being the MySmartBlinds and the other being a Texecom alarm system, both of which I have to run through homebridge with some pretty cumbersome flows/virtual switches/automation setup to get it working. I was wondering how long these things take to write, as I would be very interested in paying to have somebody write the remaining two plugins I need to get my home fully automated so to speak. Apologies if it isn't something you're interested in, but I have approached the subject on the Homey forums and not had any replies, the GitHub community seems the next logical approach. Thanks in advance

**Comment by apexad** [comment]
@nigetech I'm not interested in porting my homebridge plugins over to another platform at this time.  As far as time to write a plugin, remember that for nearly all of us, we've got full time jobs and this is just a hobby.

**Comment by nigetech** [comment]
No problems at all. Cheers

**Comment by apexad** [comment]
Once #13 is complete (which as of this writing should be the next week or so), I plan to work on this feature.  Hoping for the end of November.

**Comment by Steveome** [comment]
First things first, thanks so much for spending the time on this and creating it, I love having my blinds in Homekit.
I would also love it to Poll status once in a while (maybe every five or ten min). I utilize some switches for control, and I also use the app because of one feature (Gradual Tilt). Of course I would love for home-kit to be able to do a gradual tilt, but that may not be possible. Thanks again for your awesome work!

**Comment by apexad** [comment]
@Steveome This is coming very soon!

**Comment by nigetech** [comment]
Legendary. Thanks in advance.
This is a big thing for me, as I use the app also for gradual tilt.

**Comment by apexad** [comment]
Hoping to wrap this up by the end of the week.

**Comment by apexad** [comment]
Alright, install v2.1.0/v2.1.1 and there is a `pollingInterval` option in the config now.  Please let me know if you see any major issues.  Yet again, this was a pretty big re-write of the code.  My own testing tells me everything is working as expected though.

The 1 major note is that I've implemented a fail-safe so that if you reach the rate limit of the auth0 API (used by MySmartBlinds) it will delay polling until they set it is allowed again.

**Comment by nigetech** [comment]
Thanks @apexad do you have any idea of what should be a recommended polling interval? I have it currently sat at 5 minutes, but have no idea if this is likely to cause issues or not. Also does the bridge then poll the blinds each time, if so reducing battery life I imagine, or is it just checking with the bridge? Thanks in advance

**Comment by apexad** [comment]
I set mine to 5 minutes too, and as of yet I have not hit the rate limit, but I only have 1 blind, so not sure what multiple blinds will do.  There will be a bunch of warnings in your logs if you do hit the rate limit (if there even is one).

It's pulling the data from API server, which I assume is updated at some interval by the bridge, so I don't believe this will reduce the battery.
